server/src/main/java/org/elasticsearch/action/bulk/BulkRequest.java

```diff
@@ -87,6 +87,9 @@ public BulkRequest(StreamInput in) throws IOException {
         requests.addAll(in.readCollectionAsList(i -> DocWriteRequest.readDocumentRequest(null, i)));
         refreshPolicy = RefreshPolicy.readFrom(in);
         timeout = in.readTimeValue();
+        for (DocWriteRequest<?> request : requests) {
+            indices.add(Objects.requireNonNull(request.index(), "'request index' must not be null"));
```

Comment:
Nit:
```suggestion
            indices.add(Objects.requireNonNull(request.index(), "request index must not be null"));
```
I don't think the single quotes add real value here so might as well remove them.

Reply:
Done.